nx-X11/programs/Xserver/hw/nxagent/Display.c

```diff
@@ -1844,22 +1842,35 @@ static FILE *nxagentLookForIconFile(char *iconName, const char *permission,
 
     if (end != NULL)
     {
+      if ((end - path) > sizeof(singlePath) - 1)
+      {
+        fprintf(stderr, "Warning: Path too long - ignored.\n");
+        path = end + 1;
+        continue;
+      }
+
       strncpy(singlePath, path, (unsigned long)(end - path));
 
-      singlePath[(unsigned long)(end - path)] = 0;
+      singlePath[(unsigned long)(end - path)] = '\0';
 
       path = end + 1;
     }
     else
     {
+      if (strlen(path) > sizeof(singlePath) - 1)
+      {
+        fprintf(stderr, "Error: Path too long.\n");
+        return NULL;
+      }
+
       strcpy(singlePath, path);
 
       breakLoop = 1;
```

Comment:
With these changes, it doesn't make sense to keep `breakLoop` around. Drop it.

Reply:
must check

Reply:
Ping.